w3sper.js/src/network/mod.js

```diff
@@ -88,8 +88,10 @@ export class Network {
     );
   }
 
-  async execute(builder) {
-    const tx = await builder.build(this);
+  async execute(tx) {
+    if (typeof tx?.build === "function") {
```

Comment:
I didn't want to include a dependency to `transaction` from here (that is not the component, and it might get renamed), it didn't feel right since the `transaction` module imports nested module of `network`, so I opted to duck typing since it's idiomatic in JS.
The alternative in order to have `instanceof` here in a more clean way is to decouple the transaction module into concrete implementation and not (e.g. `Transfer` and `BasicTransfer` in a different module than the rest); however at that point I need to add a `build` method to `BasicTransfer` that basically throws.

Reply:
Fair enough